src/tree/param.h

```diff
@@ -186,7 +186,7 @@ struct TrainParam : public dmlc::Parameter<TrainParam> {
         .describe("gpu to use for single gpu algorithms");
     DMLC_DECLARE_FIELD(n_gpus)
         .set_lower_bound(-1)
-        .set_default(1)
+        .set_default(-1)
```

Comment:
Changing the default number of gpus is a big change with significant performance implications. Lets talk more about why you want to do this. The multi-gpu version can be slower depending on the input.

Reply:
OK, I will leave it unchanged.